pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -540,6 +540,8 @@ protected void doTakeSnapshot() {
       }
     }
 
+    _serverMetrics.setValueOfTableGauge(_tableNameWithType, ServerGauge.UPSERT_VALID_DOC_ID_SNAPSHOT_COUNT,
```

Comment:
Dumb question: how is this metric going to be used? like what does it tell us?

Why is this metric defined at table level rather than a partition level ?